app/controllers/hello_controller.rb

```diff
@@ -2,6 +2,7 @@ class HelloController < ApplicationController
   def index
     @picture = current_user.picture_url
     @user_name = current_user.full_name
-    @posts = Post.all
+    posts = Post.all.sort_by { |post| post[:created_at] }
+    @recent_posts = posts.reverse[0..4]
```

Comment:
we can actually do all of this in the database!

so, this isn't bad right now at all, but if we have 1million postsssss then we're going to load all 1million of them out of the database and then sort them and then take the first 5. That will probably be really slow! :(

We can do the same thing using the power of our database which is _much_ better at sorting and limiting our data. Try this out in your rails console!

```ruby
Post.all.order(created_at: :desc).limit(5)
```

That should generate a SQL query that will reduce our results to 5 sorted in a specific order before sending them back to our controller.